mupl.py

```diff
@@ -13,23 +17,199 @@
+from mupl.utils.config import config, RATELIMIT_TIME, root_path, VERBOSE, translate_message
+import mupl.utils.imagemagick_check as img_check
 
 logger = logging.getLogger("mupl")
+height_max = 10000
+
+def cup_images(input_images, output_folder, extension, path, allow_ext):
+    file_list = sorted([f for f in os.listdir(path) if f.lower().endswith(tuple(allow_ext))], key=lambda x: [int(c) if c.isdigit() else c for c in re.split(r'(\d+)', x)])
+
+    count = 1
 
+    for filename in file_list:
+        base, ext = os.path.splitext(filename)
+        new_filename = f"{base}__{ext}"
+        os.rename(os.path.join(path, filename), os.path.join(path, new_filename))
 
-def get_zips_to_upload(names_to_ids: "dict") -> "Optional[List[FileProcesser]]":
+    file_list = sorted([f for f in os.listdir(path) if f.lower().endswith(tuple(allow_ext))], key=lambda x: [int(c) if c.isdigit() else c for c in re.split(r'(\d+)', x)])
+
+    for filename in file_list:
+        base, ext = os.path.splitext(filename)
+        new_filename = f"{count:02d}{ext}"
+        os.rename(os.path.join(path, filename), os.path.join(path, new_filename))
+        count += 1
+    
+    os.makedirs(output_folder, exist_ok=True)
+    output_filename = os.path.join(output_folder, f"0.{extension}")
+    command = ["magick", "convert", "-quality", "100", "-crop", "50000x5000", f"{path}\\*.*", output_filename]
+    subprocess.run(command, check=True)
+    
+    for image_file in os.listdir(path):
+        image_path = os.path.join(path, image_file)
+        if os.path.isfile(image_path):
+            os.remove(image_path)
+    
+    output_files = [f for f in os.listdir(output_folder) if f.lower().endswith(tuple(allow_ext))]
+    for image in output_files:
+        output_pathfile = os.path.join(output_folder, image)
+        shutil.move(output_pathfile, path)
+        
+    count = 1
+
+    # Obtenha os arquivos da pasta path e renomeie-os
+    file_list = sorted([f for f in os.listdir(path) if f.lower().endswith(tuple(allow_ext))], key=lambda x: [int(c) if c.isdigit() else c for c in re.split(r'(\d+)', x)])
+    count = 1
+    for filename in file_list:
+        base, ext = os.path.splitext(filename)
+        new_filename = f"{count:02d}{ext}"
+        os.rename(os.path.join(path, filename), os.path.join(path, new_filename))
+        count += 1
+
+    os.rmdir(output_folder)
+
+
+def check_images(path, allow_ext):
+    image_files = [f for f in os.listdir(path) if f.lower().endswith(tuple(allow_ext))]
+    input_images = [os.path.join(path, image) for image in image_files]
+    output_folder = os.path.join(path, "temp")
+    
+    cup = False
+    try:
+        cup = any(int(subprocess.check_output(["magick", "identify", "-format", "%h", image]).decode("utf-8").strip()) > height_max for image in input_images)
+    except:
+        pass
+
+    if cup:
+        cup_images(input_images, output_folder, "jpg", path, allow_ext)
```

Comment:
The cropped images will always be converted to jpg as the extension argument is hardcoded.
Other than that, cropping/converting the entire folder for when it could be just one image seems suboptimal, but I don't know if there is a quick way to list all images exceeding an attribute in python. I would have suggested using `-crop 100%x50%` and recursion if it didn't shit the bed with webp and gif.

Reply:
I switched to cropping images only if they exceed 10000px, and I changed it to use Pillow instead of ImageMagick.